src/spoof-message.js

```diff
@@ -0,0 +1,117 @@
+const crypto = require("crypto");
+const axios = require("axios");
+require("./env");
+const qs = require('qs');
+
+
+// NOTES: bolt's App object requires valid credentials to initialize, so .env must contain valid credentials
+
+function calculateSignature(
+  timestampHeader, // Get it from the request header x-slack-request-timestamp
+  contentType, // Get it from the request header content-type
+  signingSecret,
+  body,
+  signatureToCompare = null, // Get it from the request header x-slack-signature
+){
+  // Based on: https://stackoverflow.com/a/70688396
+    
+  let rawBody;
+  if (contentType?.toLocaleLowerCase() === 'application/x-www-form-urlencoded') {
+    // Slash commands are sent in this content type
+    rawBody = qs.stringify(rawBody,{ format:'RFC1738' })
+  } else {
+    const stringBody = JSON.stringify(body);
+    rawBody = stringBody.replace(/\//g, '\\/').replace(/[\u007f-\uffff]/g, (c) => `\\u${  (`0000${  c.charCodeAt(0).toString(16)}`).slice(-4)}`);
+  }
+  const basestring = ['v0', timestampHeader, rawBody].join(':');
+  const calculatedSignature = `v0=${  crypto.createHmac('sha256', signingSecret).update(basestring).digest('hex')}`;
+  if (signatureToCompare) {
+    // still broken
+    const calculatedSignatureBuffer = Buffer.from(calculatedSignature, 'utf8');
+    const requestSignatureBuffer = Buffer.from(signatureToCompare, 'utf8');
+    const valid = crypto.timingSafeEqual(calculatedSignatureBuffer, requestSignatureBuffer)
+    return valid
+  }
+  return calculatedSignature;
+};
+
+function spoofMessage(
+  msg,
+  token = "52YK4N4phElR07An4dC59yMR", // optional
+  teamID = "TUMHGJ48G", // optional
+  channel = "CUM5CUN5A", // optional
+  apiAppID = "A02N5BNE13L", // optional
+  userID = "UUPDE22F9", // optional
+  eventID = "Ev03HC9TGA86", // optional
+  authUserID = "U02N2D5FRB7", // optional
```

Comment:
Are these tokens just made up, as samples? Just confirming that they aren't real values that could lead to a compromise. 😬